apps/comps/e2e/utils/rpc-client-helpers.ts

```diff
@@ -80,3 +158,456 @@ export async function createTestRpcClient(
     },
   });
 }
+
+/**
+ * Create a test admin RPC client with optional custom headers
+ *
+ * @param options - Optional configuration
+ * @param options.apiKey - API key to include in Authorization header
+ * @param options.headers - Custom headers to include
+ */
+export async function createTestAdminRpcClient(options?: {
+  apiKey?: string;
+  headers?: Record<string, string>;
+}): Promise<RouterClient<typeof adminRouter>> {
+  const headers = { ...options?.headers };
+
+  // Add Authorization header if apiKey is provided
+  if (options?.apiKey) {
+    headers.authorization = `Bearer ${options.apiKey}`;
+  }
+
+  return createRouterClient(adminRouter, {
+    context: {
+      cookies: createMockCookies("dummy-token"),
+      headers: createMockHeaders(headers),
+      params: createMockParams(),
+      adminService,
+      boostBonusService,
+      userService,
+      competitionService,
+      competitionRepository,
+      agentService,
+      arenaService,
+      partnerService,
+      balanceService,
+      portfolioSnapshotterService,
+      rewardsService,
+      logger: createLogger("TestAdminRpcClient"),
+    },
+  });
+}
+
+/**
+ * Assert that a promise throws an error with a specific error code
+ * Useful for testing RPC error responses
+ */
+export async function assertRpcError<TOutput>(
+  promise: Promise<TOutput>,
+  errorCode: string,
+  options?: {
+    messageContains?: string | string[];
+  },
+): Promise<void> {
+  const result = expect(promise).rejects;
+  await result.toThrow();
+  await result.toMatchObject({
+    code: errorCode,
+  });
+  if (options?.messageContains) {
+    const messageContainsArray = Array.isArray(options.messageContains)
+      ? options.messageContains
+      : [options.messageContains];
+
+    for (const messageContains of messageContainsArray) {
+      await result.toMatchObject({
+        message: expect.stringContaining(messageContains),
+      });
+    }
+  }
+}
+
+/**
+ * Register a user and agent using admin RPC client and return user/agent info
+ */
+export async function registerUserAndAgentAndGetClient({
```

Comment:
We already have `registerUserAndAgentAndGetClient` from the `test-helpers` package and `registerUserAndAgentAndGetRpcClient` from the comps app.  I'd rather not add yet a third variant.  Can we use the new `registerUserAndAgentAndGetRpcClient` instead?  Or at least extract out a common core between this and that one?  Also if we do need to add something new here, we should probably either do it in the comps app's test-helpers file alongside `registerUserAndAgentAndGetRpcClient`, or move that helper over here, so that they all stay together in either case.

Reply:
Hm, I see now two implementations `registerUserAndAgentAndGetClient`:
- in `hest-helpers` - it uses `ApiClient`.
- in `comps` (the new one) - it uses the `adminRpcClient`.

The former one is basically deprecated in favor of the latter, right? When we migrated all APIs to oRPC, we can delete the `ApiClient` version in `test-helpers`.

Does that make sense?

Reply:
there's also `registerUserAndAgentAndGetRpcClient` (note the "Rpc" in the name) in the comps app, which is very similar to what is being added here.  That's the one that I think we should try to unify some shared implementation with.  The old `registerUserAndAgentAndGetClient` should indeed go away once all uses of the old api have been removed